core/src/main/scala/org/apache/spark/rdd/ZippedPartitionsRDD.scala

```diff
@@ -134,6 +143,7 @@ private[spark] class ZippedPartitionsRDD4
   extends ZippedPartitionsBaseRDD[V](sc, List(rdd1, rdd2, rdd3, rdd4), preservesPartitioning) {
 
   override def compute(s: Partition, context: TaskContext): Iterator[V] = {
+    tryPrepareChildren()
```

Comment:
instead of duplicating this, can we make `compute` in the base class `final`, and have it call a `protected def doCompute` that the subclasses override?